Dangerfile

```diff
@@ -31,20 +30,45 @@ actual_reviewers = reviews.map { |u| u['user'] }
 reviewers = requested_reviewers + actual_reviewers
 pr_reviewers = reviewers.map { |u| u['login'] }
 # Official reviewer
-official_reviewer = %w[mpermperpisang mmpisang mpermper321]
+official_reviewers = %w[mpermperpisang mmpisang mpermper321]
+# File contributor reviewer
+commit_file_reviewers = []
+file_changed_list = github.api.pull_request_files(repo, pr_num)
+file_changed_name = file_changed_list.map { |u| u['filename'] }
+
+file_changed_name.map do |u|
+  commit_user_list = github.api.commits(repo, path: u.to_s)
+  commit_file_reviewers += commit_user_list.map { |c| c['commit']['author']['name'] }
+end
 
 # If reviewer not include official reviewer
-unless official_reviewer.any? { |x| pr_reviewers.include?(x) }
-  official_reviewer.delete(github.pr_author)
-  review_requests.request(official_reviewer.sample(1))
+official_reviewers.delete(github.pr_author)
+
+unless official_reviewers.any? { |x| pr_reviewers.include?(x) }
+  @official_sample = official_reviewers.sample(1)
+
+  review_requests.request(@official_sample)
+end
+
+# If reviewer not include file contribute reviewer
+regex = commit_file_reviewers.uniq.grep(/ /).to_s.gsub(/"|]|\[|\\/, '')
+
+commit_file_reviewers.delete(github.pr_author)
+commit_file_reviewers.delete(@official_sample)
+commit_file_reviewers.delete(regex)
+
+unless commit_file_reviewers.any? { |x| pr_reviewers.include?(x) }
+  sample = file_changed_name.length > commit_file_reviewers.length ? commit_file_reviewers.length : file_changed_name.length
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mpermperpisang/golang-automation/blob/374beeb0459350267d4f9e26e212f51e8f8fb470/Dangerfile#L61"></span>Line is too long. [124/120]</td>
    </tr>
  </tbody>
</table>